.gitpod.yml

```diff
@@ -0,0 +1,90 @@
+# Learn more about this file at https://www.gitpod.io/docs/references/gitpod-yml
+tasks:
+  - name: Service Server
+    # Get Web API urls for use in Next build
+    before: |
+      eval $(gp env -e GITPOD_API_URL=$(gp url 3000))
+      eval $(gp env -e GITPOD_WEB_URL=$(gp url 8000))
+      eval $(gp env -e GITPOD_MICROSERVICE_URL=$(gp url 8443))
+    init: |
+      npm install
+      docker-compose --env-file config/env.gitpod build
+    # GitPod workspace urls are not stable, we have to look up urls and build again
+    command: |
+      eval $(gp env -e GITPOD_API_URL=$(gp url 3000))
+      eval $(gp env -e GITPOD_WEB_URL=$(gp url 8000))
+      eval $(gp env -e GITPOD_MICROSERVICE_URL=$(gp url 8443))
+      docker-compose --env-file config/env.gitpod build
```

Comment:
I think that we should do this using variable overriding, see https://docs.docker.com/compose/environment-variables/#set-environment-variables-with-docker-compose-run, meaning:

```sh
eval $(gp env -e API_URL=$(gp url 3000))
eval $(gp env -e WEB_URL=$(gp url 8000))
eval $(gp env -e API_HOST=$(gp url 8443))
docker-compose --env-file config/env.gitpod -e WEB_URL -e API_HOST -e API_URL build
```

I'm pretty sure (untested!) that this will use the values you just evaluated in the current shell to pass them down, overriding what's in the `env` file.  If that works, you could remove the definitions from `config/env.gitpod` and replace with a comment about how they are defined in this process, pointing to this file.

Doing this should clean up the code changes you had to make below to compensate for things like the Post service API URL.

You'd do the same below on line 19 to pass these variables through to the containers when running `up`.

cc @manekenpix, @raygervais who can help check my logic here.

Reply:
oh, that would me much better, I was just following how production forward env var.
I'm trying to get this merged today but won't be available until later today.
If you can, could you please test it on gitpod and push commit to this PR if you want to?

Reply:
I'm not able to test this today, but maybe others can?  Do you need this for Hacktoberfest?  If so, we can land a version of it and iterate; otherwise I'd like to keep it open and do this properly.

Reply:
I'll need this for Hacktoberfest. Cindy seems to be busy right now. 
But I think can fix your change requests today and merge it.
Give me a few hours now, I'll be back soon.

Reply:
If you need this in today, let's drop the changes you made to the next/front end in `src/web/*`, and only include the gitpod specific stuff (yml, docker, env files).  Then we can follow-up to get it working properly in later PRs.  I think this is close, but I don't want to mess around with our dev/production build and frontend in order to support gitpod.  We should do all this monkey-patching in the gitpod config.  I'll check in on this later today and help you get it in.